src/PortfolioPlanning/Common/Services/BacklogConfigurationDataService.ts

```diff
@@ -30,12 +31,17 @@ export class BacklogConfigurationDataService {
 
         const projectEfforFieldRefName =
             projectBacklogConfiguration.backlogFields &&
-                projectBacklogConfiguration.backlogFields.typeFields[BacklogConfigurationDataService.EffortTypeField]
+            projectBacklogConfiguration.backlogFields.typeFields[BacklogConfigurationDataService.EffortTypeField]
                 ? projectBacklogConfiguration.backlogFields.typeFields[BacklogConfigurationDataService.EffortTypeField]
                 : null;
 
         const portfolioLevelsData = this.getPortfolioLevelsData(projectBacklogConfiguration);
 
+        const portfolioWorkItemTypes = Object.keys(portfolioLevelsData.backlogLevelNamesByWorkItemType);
+        const workItemTypeMappedStates: WorkItemTypeStateInfo[] = projectBacklogConfiguration.workItemTypeMappedStates.filter(
+            item => portfolioWorkItemTypes.indexOf(item.workItemTypeName.toLowerCase()) !== -1
+        );
```

Comment:
Alternatively, you could do a `forEach` on `projectBacklogConfiguration.workItemTypeMappedStates` and check if the work item type is a key of the `portfolioLevelsData.backlogLevelNamesByWorkItemType` map.